### PR TITLE
refactor: Move POST /users and POST /users/:id to invitation controller (no-changelog)

### PR DESCRIPTION
**Pull request by RicardoE105**
Built on top of #7714 

This PR:

1. Create invitation controller
2. Move `POST /users` to `POST /invitations` and move related test to `inviations.api.tests`
3. Move `POST /users/:id` to `POST /invitations/:id/accept` move related test to `inviations.api.tests`
4. Adjust FE to use new endpoints

### CODE REVIEW

File: packages/cli/src/controllers/invitation.controller.ts [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+			{ userShells: createUsers },
+		);
+
+		return emailingResults;
+	}
+
+	/**
+	 * Fill out user shell with first name, last name, and password.
+	 */
+	@NoAuthRequired()
```
Comment (RicardoE105) [inline packages/cli/src/controllers/invitation.controller.ts:238]: For some reason this decorator is not working and the route is always authenticated. @netroy any idea what might be happening?

Comment (netroy) [reply packages/cli/src/controllers/invitation.controller.ts:238]: This is bug in the legacy auth middleware, that prevents requests from reaching these routers, and since that middleware is applied globally, we can't exclude routes because `req.route` does not exist in that middleware.
This is why we've had [this check](https://github.com/n8n-io/n8n/blob/master/packages/cli/src/middlewares/auth.ts#L92) in place.
I suggest that we update [these lines](https://github.com/n8n-io/n8n/blob/master/packages/cli/src/middlewares/auth.ts#L65-L66) to match the new URL.
Once we have all the old route out of the main express app, and into dedicated routers, we can then get rid of the global auth middleware, and all the checks like these.

File: packages/cli/src/controllers/invitation.controller.ts [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+
+		const role = await this.roleService.findGlobalMemberRole();
+
+		if (!role) {
+			this.logger.error(
+				'Request to send email invite(s) to user(s) failed because no global member role was found in database',
+			);
+			throw new InternalServerError('Members role not found in database - inconsistent state');
+		}
+
```
Comment (netroy) [inline packages/cli/src/controllers/invitation.controller.ts:113]: Can we please create a new `InvitationService` and move the code after this line into a method `inviteUsers` in that service? 
Then we can use that service directly in external hooks.

File: packages/cli/src/controllers/invitation.controller.ts [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+export class InvitationController {
+	constructor(
+		private readonly config: Config,
+		private readonly logger: Logger,
+		private readonly internalHooks: IInternalHooksClass,
+		private readonly externalHooks: IExternalHooksClass,
+		private readonly userService: UserService,
+		private readonly roleService: RoleService,
+		private readonly mailer: UserManagementMailer,
+		private readonly postHog?: PostHogClient,
```
Comment (netroy) [inline packages/cli/src/controllers/invitation.controller.ts:40]: we need to get rid of this optional dependency stuff. then we can use `Container.get`, instead of having to manually instantiate this class everywhere.
If this PR isn't urgent, I'd like to make that change in another PR, before the end of this day.

**Review by netroy — COMMENTED** [review]
looks good so far 👍🏽

File: packages/cli/src/controllers/invitation.controller.ts (hunk 3 above)
Comment (RicardoE105) [reply packages/cli/src/controllers/invitation.controller.ts:40]: This is not urgent!

Comment (RicardoE105) [reply packages/cli/src/controllers/invitation.controller.ts:40]: It think postHog is optional because we do not inject it in the tests for some reason. In we inject it on the test, we can get rid of it

**Comment by netroy** [comment]
closing in favor of #7726
